4iprocessing/generate_initial_config.py

```diff
@@ -0,0 +1,144 @@
+parser.add_argument('--output_path', type=str, required=True, help="final alignment outputs to specify in yaml file")
+parser.add_argument('--output_yaml_dir', type=str, required=True, help="final alignment outputs to specify in yaml file")
+
+
+
+def sort_rounds(rounds):
+    # function to sort rounds
+    Orginal_numbered_list = []
+    for i in range(len(rounds)):
+        Orginal_numbered_list.append(int(re.search(r'\d+', rounds[i]).group())) 
+    print(f"original list is {Orginal_numbered_list}")
+    sorted_list = sorted(Orginal_numbered_list)
+    #print(f"sorted list is {sorted_list}")
+    final_sorted_list=[]
+    for num in set(sorted_list):
+        #index = Orginal_numbered_list.index(num)
+        indexs=[i for i,val in enumerate(Orginal_numbered_list) if val==num]
+        print(f"for num {num}, indexs is {indexs}")
+        for k in range(len(indexs)):
+            idx = indexs[k]
+            print(idx)
+            final_sorted_list.append(rounds[idx])
+    return final_sorted_list
+
+
+def check_regex_for_changing_ref_channel(round_name):
+    # return True for 20x timelapse and round1, return False otherwise. 
+    # helps deal with differing regex scheming
+    if bool(re.search('time',round_name, re.IGNORECASE)):
+        return True
+    elif bool(re.search('Round', round_name, re.IGNORECASE)):
+        round_number= int(re.search(r'\d+', round_name).group())
+        if round_number==1:
+            return True
+        else:
+            return False
+    else:
+        return False
+
+def get_scenes_to_toss(reader):
+    # get scenes to toss as assesed by if reader can get dimensions of that scene
+    scenes_to_toss=[]
+    for scene in reader.scenes:
+        reader.set_scene(scene)
+        si =reader.current_scene_index
+        try: 
+            dims = reader.dims
+        except:
+            scenes_to_toss.append(si+1)
+
+    return scenes_to_toss
+
+if __name__ == '__main__':
+    args= parser.parse_args()
+
+    for bdir in args.input_dirs: # for each input directory
+        barcode = Path(Path(bdir)).name
+        print(barcode)
+        config ={}
+        config['Data']=[]
+        scope_list = [x for x in os.listdir(bdir) if 'ZSD' in x]
+        for scope in scope_list: # for each scope
+            pdir = bdir + os.sep + scope
+            round_list = [x for x in os.listdir(pdir) if bool(re.search('Time|Round [0-9]+',x,re.IGNORECASE))&(Path(x).stem==Path(x).name)]
+            round_list = sort_rounds(round_list)
+            print("round list is {}".format(round_list))
+            for round_num in round_list:
+                ppath = os.path.join(pdir, round_num)
+                czi_list = [x for x in os.listdir(ppath) if ('.czi' in x)&bool(re.search('20x',x,re.IGNORECASE))]
```

Comment:
we will need to include support for ome.tiffs as well, since those are the aligned files

Reply:
also just took care of this one